service/worker/batcher/workflow.go

```diff
@@ -235,6 +247,24 @@ func validateParams(params BatchParams) error {
 			return fmt.Errorf("must provide signal name")
 		}
 		return nil
+	case BatchTypeUpdateOptions:
+		if params.UpdateOptionsParams.WorkflowExecutionOptions == nil {
+			return fmt.Errorf("must provide UpdateOptions")
+		}
+		if params.UpdateOptionsParams.UpdateMask == nil {
+			return fmt.Errorf("must provide UpdateMask")
+		}
+		if override := params.UpdateOptionsParams.WorkflowExecutionOptions.VersioningOverride; override != nil {
+			if override.GetBehavior() == enumspb.VERSIONING_BEHAVIOR_PINNED &&
+				override.GetDeployment() == nil {
+				return fmt.Errorf("must provide Deployment if Behavior is 'PINNED'")
+			}
```

Comment:
let's also add a check to make sure behavior is always set so user is not passing an empty override?

Reply:
yeah, probably I should make a `validateBehavior` fn since I'm doing this in a few different places now

Reply:
done